### ISSUE TITLE
Blobfuse slower performance with MSI compared to SAS

### ISSUE THREAD

**Issue by trapeznikov** [Azure/azure-storage-fuse#548]
### Which version of the blobfuse was used?
1.3.6

### Which OS (please include version) are you using?
RHEL 7.8

### What problem was encountered?
Same version of blobfuse, same mount command, experience slower read/writes when mounted with MSI compared to SAS.

#### Test with SAS
```
date; for i in 1 2 3; do time dd if=/dev/zero of=perf_test_blob_sas count=10000 bs=1M; done
Tue Dec 29 14:53:49 EST 2020
10000+0 records in
10000+0 records out
10485760000 bytes (10 GB) copied, 66.4274 s, 158 MB/s

real	1m6.615s
user	0m0.046s
sys	0m8.809s
10000+0 records in
10000+0 records out
10485760000 bytes (10 GB) copied, 62.6406 s, 167 MB/s

real	3m56.766s
user	0m0.032s
sys	0m10.016s
10000+0 records in
10000+0 records out
10485760000 bytes (10 GB) copied, 63.8951 s, 164 MB/s

real	4m4.883s
user	0m0.042s
sys	0m10.147s
```

#### Test with MSI
```
date; for i in 1 2 3; do time dd if=/dev/zero of=perf_test_blob_msi count=10000 bs=1M; done
Tue Dec 29 14:04:37 EST 2020
10000+0 records in
10000+0 records out
10485760000 bytes (10 GB) copied, 144.176 s, 72.7 MB/s

real	2m24.370s
user	0m0.035s
sys	0m8.738s
10000+0 records in
10000+0 records out
10485760000 bytes (10 GB) copied, 139.404 s, 75.2 MB/s

real	5m38.758s
user	0m0.038s
sys	0m10.112s
10000+0 records in
10000+0 records out
10485760000 bytes (10 GB) copied, 148.069 s, 70.8 MB/s

real	6m36.456s
user	0m0.030s
sys	0m10.039s
```
### Have you found a mitigation/solution?
No

### By default, blobfuse logs errors to syslog. If this is relevant, is there anything in the syslog that might be helpful?
Wasn't able to find anything related to the issue in logs or debug output

### If relevant, please share your mount command.
blobfuse /RW/staging-edp-dev --tmp-path=/mnt/resource/staging-edp-dev_fsdev --container-name=staging-edp-dev --file-cache-timeout-in-seconds=0 -ogid=933389 -o allow_other

**Comment by NaraVen** [comment]
yes MSI is slower and that is expected but you can get some improvements by tweaking your cache related command line params and taking advantage of caching. You can add the below options while mounting.
 --file-cache-timeout-in-seconds=3600 -o attr_timeout=240 -o entry_timeout=240 -o negative_timeout=120

**Comment by trapeznikov** [comment]
Hi @NaraVen, caching will help with reading speed, right? But based on my testing even writing is about twice as long with MSI.

**Comment by trapeznikov** [comment]
Please see the testing results with the cache parameters tuned. As expected, read speed was improved (second and third run), but write still takes twice as long (first run) with MSI.

MSI:
```
date; for i in 1 2 3; do time dd if=/dev/zero of=perf_test_blob_msi-with-cache-timeout count=10000 bs=1M; done
Tue Jan  5 14:54:42 EST 2021
10000+0 records in
10000+0 records out
10485760000 bytes (10 GB) copied, 103.496 s, 101 MB/s
 
real        1m43.676s
user       0m0.038s
sys         0m8.884s
10000+0 records in
10000+0 records out
10485760000 bytes (10 GB) copied, 103.303 s, 102 MB/s

real        2m2.519s
user       0m0.021s
sys         0m9.896s
10000+0 records in
10000+0 records out
10485760000 bytes (10 GB) copied, 102.434 s, 102 MB/s

real        2m5.032s
user       0m0.042s
sys         0m9.967s
 ```

SAS:
```
date; for i in 1 2 3; do time dd if=/dev/zero of=perf_test_blob_sas-with-cache-timeout count=10000 bs=1M; done
Tue Jan  5 15:03:41 EST 2021
10000+0 records in
10000+0 records out
10485760000 bytes (10 GB) copied, 47.9363 s, 219 MB/s

real        0m48.256s
user       0m0.038s
sys         0m8.943s
10000+0 records in
10000+0 records out
10485760000 bytes (10 GB) copied, 47.5373 s, 221 MB/s

real        1m58.409s
user       0m0.040s
sys         0m10.108s
10000+0 records in
10000+0 records out
10485760000 bytes (10 GB) copied, 47.7078 s, 220 MB/s

real        2m1.690s
user       0m0.037s
sys         0m10.284s
```

**Comment by NaraVen** [comment]
We will improve the perf, currently caused by date comparison and push it in the next release.

**Comment by elabbarw** [comment]
Any update on this please?

**Comment by NaraVen** [comment]
We will update you when we decide on a release date.

**Comment by vibhansa-msft** [comment]
Fixed in 1.3.7, kindly upgrade.
